### DISCUSSION TITLE
withOne eager-loading param

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/cms#8705] [Ideas]
Element queries should have a `withOne` param, which would work similar to `with`, except that only the first related element would be loaded for each parent element, and those eager-loaded elements wouldn’t need to be accessed using the array syntax (so `entry.fieldHandle` rather than `entry.fieldHandle[0]`).

**Comment by putyourlightson** [comment]
I submitted pull request https://github.com/craftcms/cms/pull/4598 as a proposed simplified solution. If you however decide that `withOne` should work similarly to `with` then I'd be happy to take a shot at that too.

**Comment by brandonkelly** [comment]
Gotta hold off until 4.0 for this, as it’s going to introduce a breaking change for element types that have implemented custom eager-loading properties, and aren’t anticipating the possibility of `withOne = true`.

**Comment by brandonkelly** [comment]
Though worth noting that as of Craft 3.3.1, it is possible to pass `limit: 1` to an eager loading criteria. Not as elegant as this idea, but at least provides a bit of a performance boost when only one element is needed.

```twig
{% set entries = craft.entries()
    .with([
        ['photo', {limit: 1}],
    ])
    .all() %}
```
